### ISSUE TITLE
Forms with streams - static content disappears after validate

### ISSUE THREAD

**Issue by mike1o1** [phoenixframework/phoenix_live_view#3039]
### Environment

* Elixir version (elixir -v): Erlang/OTP 26 [erts-14.1] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit]

Elixir 1.15.6 (compiled with Erlang/OTP 26)
* Phoenix version (mix deps): phoenix 1.7.2

* Phoenix LiveView version (mix deps): phoenix_live_view 0.19.0
* Operating system: Mac and Ubuntu
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, Safari
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior
I'm relatively new to Elixir and Phoenix, so I hope I'm getting my terminology correct.

What I'm experiencing is if I have a stream of forms, but render some of the changeset data outside of the form, then the content disappears after a phx-validate event happens.

I was able to reproduce in the `todo_trek` [app](https://github.com/chrismccord/todo_trek) repo by rendering some of the static content, for example:

```heex
<.simple_form
  for={form}
  phx-change="validate"
  phx-submit="save"
  phx-value-id={form.data.id}
  phx-target={@myself}
  class="min-w-0 flex-1 drag-ghost:opacity-0"
>
  <div> <-- This div is new
    <p>
      Static: <%= form.data.title %> 
    </p>
  </div>
  <div class="flex">
    <button
      :if={form.data.id}
      type="button"
      phx-click={JS.push("toggle_complete", target: @myself, value: %{id: form.data.id})}
      class="w-10"
    >
...
```

Even if I bring that content outside the form, it still clears out as soon as a validate event is handled. Unfortunately, that validate event takes place either based on user input _or_ when the socket reconnects, which can leave the user with a blank page, basically (for the static content at least).

Everything renders fine on page load or normal page navigation, however if I attempt to type in the input field, then the same value outside of the input fields are wiped out.

### Expected behavior
I'd expect the static content to not disappear. The above behavior happens both in dev and production, and is easily reproduced on a phone which seems to have more aggressive websocket closing than desktop. Also, each time I do a deploy, every connected page sees this behavior.

### One-line File

https://gist.github.com/mike1o1/1b1e4b93598ad416084152e69eaeb76a

If you run this gist, then you'll see that as soon as one of the inputs is changed, the dynamic content of the stream is empty, even for fields that aren't connected to an input (such as the description).

I'm trying to implement a sort of "inline" edit functionality using only client-side functionality, by rendering both the static content of a form, as well as an editable, with the user able to toggle between the two by hitting an "Edit" button and using JS commands to hide the static content and show the editable form, but this is a pretty big blocker.

**Comment by josevalim** [comment]
Can you please update to the latest Phoenix live view and let us know if the issue still happens?

**Comment by SteffenDE** [comment]
Your single file reproduction does not fit to your description of the problem. In your validate handler you insert a new todo item that has an empty content and description, thus replacing the existing item in the stream. Is is correct that the content and description render as `""` after validate. Could you clarify the problem or adjust the reproduction? To me it sounds like your initial problem could be fixed on main, so please also try this:

```
{:phoenix_live_view, github: "phoenixframework/phoenix_live_view", branch: "main", override: true}
```

**Comment by mike1o1** [comment]
Ok, I was able to track this one down to user error, and not a bug in the framework. The issue I ran into is I was trying to access the `data` property on the `%Phoenix.HTML.Form{}`, not realizing that it wouldn't have the latest changes to the struct after validation is called.

Instead, I need to use the Access behavior, and access the value by doing `item[:property].value`. When the page is first loaded, the `data` property on the form contains the values, but after validation (either through user input or socket reconnect), the data is emptied, which is why it appeared that the values were empty.

Thanks for helping point in the right direction!
